terraso_backend/apps/graphql/schema/story_maps.py

```diff
@@ -103,3 +108,117 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
                 model_name=StoryMap.__name__, operation=MutationTypes.DELETE
             )
         return super().mutate_and_get_payload(root, info, **kwargs)
+
+
+class StoryMapMembershipSaveMutation(BaseAuthenticatedMutation):
+    model_class = Membership
+    membership = graphene.Field(CollaborationMembershipNode)
+
+    class Input(BaseSaveInput):
+        story_map_id = graphene.String(required=True)
+        story_map_slug = graphene.String(required=True)
+
+    @classmethod
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+
+        if kwargs["user_role"] != ROLE_CONTRIBUTOR:
+            logger.info(
+                "Attempt to save a Membership, but user has no permission",
+                extra=kwargs,
```

Comment:
question: does kwargs contain information about the user and membership list? Should the words "story map" appear in the error message?